packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,244 @@
+import React, {
+  createContext, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { setPassportClientId, track } from '@imtbl/metrics';
+import { Passport } from '../Passport';
+import { PassportModuleConfiguration, UserProfile } from '../types';
+import { Provider } from '../zkEvm/types';
+
+type PassportContextType = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+  error: Error | null;
+  accounts: string[];
+  profile: UserProfile | null;
+  passportProvider: Provider | null;
+  login: (withoutWallet?: boolean) => Promise<void>;
+  logout: () => Promise<void>;
+  loginCallback: () => Promise<void>;
+  linkExternalWallet: Passport['linkExternalWallet'];
+};
+
+const PassportContext = createContext<PassportContextType | null>(null);
+
+type PassportProviderProps = {
+  config: PassportModuleConfiguration;
+  children: React.ReactNode;
+};
+
+export function ReactProvider({ children, config }: PassportProviderProps) {
+  setPassportClientId(config.clientId);
+  track('passport', 'reactProviderInitialised');
+  const [isLoggedIn, setLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [accounts, setAccounts] = useState<string[]>([]);
+  const [profile, setProfile] = useState<UserProfile | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+  const [passportProvider, setPassportProvider] = useState<Provider | null>(null);
+  const passportInstance = new Passport(config);
+
+  const getPassportProvider = () => {
+    if (passportProvider) return passportProvider;
+    const p = passportInstance.connectEvm();
+    setPassportProvider(p);
+    return p;
+  };
+
+  const v = useMemo(() => ({
+    passportInstance,
+    passportProvider,
+    isLoggedIn,
+    isLoading,
+    accounts,
+    profile,
+    error,
+    login: async (withoutWallet: boolean = false) => {
+      try {
+        setError(null);
+        setIsLoading(true);
+        if (withoutWallet) {
+          const p = await passportInstance.login();
+          setProfile(p);
+          setLoggedIn(true);
+        } else {
+          const provider = getPassportProvider();
+          const acc = await provider.request({ method: 'eth_requestAccounts' });
+          setLoggedIn(true);
+          setAccounts(acc);
+        }
+      } catch (e) {
+        setError(e as Error);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    logout: async () => {
+      try {
+        setError(null);
+        setIsLoading(true);
+        await passportInstance.logout();
+        setPassportProvider(null);
+        setLoggedIn(false);
+        setAccounts([]);
+        setProfile(null);
+      } catch (e) {
+        setError(e as Error);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    loginCallback: async () => {
+      await passportInstance.loginCallback();
+    },
+    linkExternalWallet: passportInstance.linkExternalWallet.bind(passportInstance),
```

Comment:
Is the `bind` call here necessary? It's not clear to me why we can't just call `linkExternalWallet` directly:
```suggestion
    linkExternalWallet: passportInstance.linkExternalWallet,
```

Reply:
`linkExternalWallet` uses `this` in its function body tho. Without the bind, it'll work?